src/renderers/webgl/WebGLPrograms.js

```diff
@@ -190,9 +194,12 @@ function WebGLPrograms( renderer, cubemaps, cubeuvmaps, extensions, capabilities
 			normalMap: !! material.normalMap,
 			objectSpaceNormalMap: material.normalMapType === ObjectSpaceNormalMap,
 			tangentSpaceNormalMap: material.normalMapType === TangentSpaceNormalMap,
-			clearcoatMap: !! material.clearcoatMap,
-			clearcoatRoughnessMap: !! material.clearcoatRoughnessMap,
-			clearcoatNormalMap: !! material.clearcoatNormalMap,
+
+			clearcoat: useClearcoat,
+			clearcoatMap: useClearcoat && !! material.clearcoatMap,
+			clearcoatRoughnessMap: useClearcoat && !! material.clearcoatRoughnessMap,
+			clearcoatNormalMap: useClearcoat && !! material.clearcoatNormalMap,
```

Comment:
@WestLangley Should we do something like this for `transmission` too? If so... what parameters depend on `material.transmision > 0`?

Reply:
```js
transmission: <float>,
transmissionMap: new THREE.Texture( <Image> ),

thickness: <float>,
thicknessMap: new THREE.Texture( <Image> ),
attenuationDistance: <float>,
attenuationTint: <Color>,
```

Reply:
Excellent! I'll do that on my next PR.

Reply:
But actually, transmission and volume are separate extensions. The last four are for volume.

Reply:
Oh! But in the context of our shader chunks...?

Reply:
They really should be separate. When `thickness` is zero, the "transmission only" code should execute.

Reply:
The first two properties are for transmission only -- or for both transmission and volume.

Reply:
When thickness is zero, the volume code should return the same answer as one would get if the transmission-only-extension were implemented. It is just slower, so it would be better to have a separate code path for transmission-only.

Reply:
>Should we do something like this for transmission too?

Yes.

Related: You can do something like this for `sheenTint` too, so the version will be incremented when `sheenTint` is changed from black to not-black.

Reply:
> Related: You can do something like this for `sheenTint` too, so the version will be incremented when `sheenTint` is changed from black to not-black.

Yep! Step by step 🤓

Reply:
> Related: You can do something like this for sheenTint too, so the version will be incremented when sheenTint is changed from black to not-black.

Would love to see that. That would make the physical exclusive properties more consistent.

Reply:
So then the logic would be like this?

```js
if ( transmision > 0 ) {

    transmission: <float>
    transmissionMap: new THREE.Texture( <Image> )

    if ( thickness > 0 ) {

        thickness: <float>
        thicknessMap: new THREE.Texture( <Image> )
        attenuationDistance: <float>
        attenuationTint: <Color>

    }

}
```

Reply:
There should be `USE_TRANSMISSION` and `USE_VOLUME`.

It is up to you if `USE_TRANSMISSION` is used to mean "transmission_without_volume" -- a.k.a. thin surface only.

Reply:
>Oh! But... in the context of our shader chunks, do thickness and attenuation affect the material if transmission is 0?

No.